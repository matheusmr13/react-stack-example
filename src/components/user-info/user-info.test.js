import React from 'react';
import { render, shallow } from 'enzyme';
import UserInfo from './index';

const user = {
	info: {
		display_name: 'My User',
		external_urls: {
			spotify: 'spotifyurl'
		},
		followers: {
			total: 65
		},
		href: 'userurl',
		images: [{
			url: 'userimageurl'
		}]
	}
};

describe('UserInfo component', () => {
	it('should render properly without user', () => {
		const entityFilter = render(<UserInfo onLogout={() => {}} />);
		expect(entityFilter).toMatchSnapshot();
	});
	it('should render properly with user', () => {
		const entityFilter = render(<UserInfo user={user} onLogout={() => {}} />);
		expect(entityFilter).toMatchSnapshot();
	});
	it('should logout user if button clicked', () => {
		const onLogout = jest.fn();
		const entityFilter = shallow(<UserInfo user={user} onLogout={onLogout} />);
		entityFilter.find('button').simulate('click');
		expect(onLogout).toHaveBeenCalled();
	});
});
