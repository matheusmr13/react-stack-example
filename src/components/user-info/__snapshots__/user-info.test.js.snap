// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo component should render properly with user 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "user-info",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "user-info__data",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "user-info__name",
            },
            "children": Array [
              Object {
                "data": "My User",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "user-info__followers",
              },
              "children": Array [
                Object {
                  "data": "65 followers",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "user-info__logout",
                },
                "children": Array [
                  Object {
                    "data": "Logout",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "user-info__followers",
            },
            "children": Array [
              Object {
                "data": "65 followers",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "user-info__logout",
              },
              "children": Array [
                Object {
                  "data": "Logout",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "user-info__name",
              },
              "children": Array [
                Object {
                  "data": "My User",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "user-info__logout",
            },
            "children": Array [
              Object {
                "data": "Logout",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "user-info__followers",
              },
              "children": Array [
                Object {
                  "data": "65 followers",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "user-info__name",
                },
                "children": Array [
                  Object {
                    "data": "My User",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "user-info__image",
            "href": "spotifyurl",
            "target": "_blank",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "alt": "My User",
                "src": "userimageurl",
              },
              "children": Array [],
              "name": "img",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "alt": undefined,
                "src": undefined,
              },
              "x-attribsPrefix": Object {
                "alt": undefined,
                "src": undefined,
              },
            },
          ],
          "name": "a",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "href": undefined,
            "target": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "href": undefined,
            "target": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "user-info__image",
          "href": "spotifyurl",
          "target": "_blank",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "alt": "My User",
              "src": "userimageurl",
            },
            "children": Array [],
            "name": "img",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "alt": undefined,
              "src": undefined,
            },
            "x-attribsPrefix": Object {
              "alt": undefined,
              "src": undefined,
            },
          },
        ],
        "name": "a",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "user-info__data",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "user-info__name",
              },
              "children": Array [
                Object {
                  "data": "My User",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "user-info__followers",
                },
                "children": Array [
                  Object {
                    "data": "65 followers",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "user-info__logout",
                  },
                  "children": Array [
                    Object {
                      "data": "Logout",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "user-info__followers",
              },
              "children": Array [
                Object {
                  "data": "65 followers",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "user-info__logout",
                },
                "children": Array [
                  Object {
                    "data": "Logout",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "user-info__name",
                },
                "children": Array [
                  Object {
                    "data": "My User",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "user-info__logout",
              },
              "children": Array [
                Object {
                  "data": "Logout",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "user-info__followers",
                },
                "children": Array [
                  Object {
                    "data": "65 followers",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "user-info__name",
                  },
                  "children": Array [
                    Object {
                      "data": "My User",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "href": undefined,
          "target": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "href": undefined,
          "target": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`UserInfo component should render properly without user 1`] = `
initialize {
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
